### PR TITLE
Fix Fluent files for Moz accounts update

### PR DESCRIPTION
**Pull request by reemhamz**
## One-line summary

Received feedback from the L10N team regarding the Fluent files we merged regarding the Mozilla accounts update coming up: https://github.com/mozilla-l10n/www-l10n/pull/350

No major string changes, just tiny fix-ups.

### Security pages
I also included tiny string changes (English only, no localization) to the following page:
- http://localhost:8000/en-US/security/third-party-software-injection/

### CODE REVIEW

**Review by alexgibson — COMMENTED** [review]
It's not clear to me which is these strings are either new or outdated. If you're changing existing translated strings, would those require a new string ID?

File: l10n/en/firefox/features/password-manager-2023.ftl
Comment (reemhamz) [reply l10n/en/firefox/features/password-manager-2023.ftl:22]: So in the original PR I created that added the new strings, I somehow (I honestly don't remember doing this specific change) updated `{ -brand-name-firefox-account }` to `{ -brand-name-firefox } account`. You can see it here: https://github.com/mozilla/bedrock/pull/13690/files#diff-29a156658b629619541921aa462c94521caa4f044fbe03d445616b3d3f65e318L17

So the L10N team wanted me to change it back to how it was since it was becoming an obsolete string anyway

File: l10n/en/firefox/features/sync-2023.ftl
Comment (reemhamz) [reply l10n/en/firefox/features/sync-2023.ftl:19]: Just changing it back to how it was before the original PR change: https://github.com/mozilla/bedrock/pull/13690/files#diff-29a156658b629619541921aa462c94521caa4f044fbe03d445616b3d3f65e318L17, similar change to the `password-manager-2023.ftl` file

File: l10n/en/firefox/products.ftl
Comment (alexgibson) [reply l10n/en/firefox/products.ftl:61]: This string looks like an old one that's well translated? I don't see it in the original PR at all?

**Review by alexgibson — APPROVED** [review]
r+

File: bedrock/firefox/templates/firefox/accounts.html
Comment (reemhamz) [inline bedrock/firefox/templates/firefox/accounts.html:56]: Just got confirmation that we won't use the Mozilla logo in the Accounts page after the change launches. Figured I'd just remove it quickly in this PR

**Review by alexgibson — APPROVED** [review]
r++

**Comment by peiying2** [comment]
@reemhamz I should have held off on the merge of https://github.com/mozilla-l10n/www-l10n/pull/350 before the updates here.

**Comment by flodolo** [comment]
Sorry for the extra work, this is due to an error on our side (the PR shouldn't have been merged).

What happened is that we already exposed the strings for translation, and there is no way to undo it (a backout won't be enough). At this point, we need to force a new translation to pick up these changes, and that requires using new IDs for these strings (with the exception of strings where only capitalization changes, i.e. `Sign In`-> `Sign in`).

Again, sorry for the mess 😢 

In the code, feel free to drop the middle version as a fallback between obsolete and new string.

**Comment by reemhamz** [comment]
@peiying2 @flodolo I updated the FTL files to add newer versions of the strings that should hopefully cancel out the older ones I created for the moz account updates. Let me know if these updates are correct and we can merge and proceed?

File: l10n/en/firefox/browsers.ftl
Comment (flodolo) [inline l10n/en/firefox/browsers.ftl:44]: Just to confirm, this is OK as is, since it's only relevant for English.

File: l10n/en/firefox/browsers/compare/safari.ftl
Comment (flodolo) [inline l10n/en/firefox/browsers/compare/safari.ftl:85]: You should remove this from the file, as it's not used anymore.

File: l10n/en/firefox/features/password-manager-2023.ftl
Comment (flodolo) [inline l10n/en/firefox/features/password-manager-2023.ftl:22]: This needs a new ID, even if it's obsolete (there are translations with a different term, and it will create warnings all over the place).

But, if it's indeed obsolete, we should leave it as is.

File: l10n/en/firefox/features/sync-2023.ftl
Comment (flodolo) [inline l10n/en/firefox/features/sync-2023.ftl:23]: This needs a new ID. As above, if it's actually unused, we should probably leave it as is and don't change it.

File: l10n/en/firefox/features/sync-2023.ftl
Comment (flodolo) [inline l10n/en/firefox/features/sync-2023.ftl:20]: This is not used anymore, it should be removed.

File: l10n/en/sub_navigation.ftl
Comment (flodolo) [inline l10n/en/sub_navigation.ftl:49]: Remove, as it's unused.

**Review by flodolo — CHANGES_REQUESTED** [review]
(I'd suggest to ask a second review from Alex, because I might be easily missing something)

File: l10n/en/firefox/features/password-manager-2023.ftl
Comment (reemhamz) [reply l10n/en/firefox/features/password-manager-2023.ftl:22]: This change was just reverting the string to how it was before the initial Moz accounts update merge. Comment with some more info mentioned here: https://github.com/mozilla/bedrock/pull/13706#discussion_r1333049850

File: l10n/en/firefox/features/sync-2023.ftl
Comment (reemhamz) [reply l10n/en/firefox/features/sync-2023.ftl:23]: Explanation for this change here: https://github.com/mozilla/bedrock/pull/13706#discussion_r1333057273

Should I still change it?

File: l10n/en/firefox/features/password-manager-2023.ftl
Comment (flodolo) [reply l10n/en/firefox/features/password-manager-2023.ftl:22]: The string that we [currently expose](https://github.com/mozilla-l10n/www-l10n/blob/master/en/firefox/features/password-manager-2023.ftl#L22C79-L22C110) has `term + account`, and [all translations](https://transvision.flod.org/string/?entity=mozilla_org/en/firefox/features/password-manager-2023.ftl:password-manager-with-a-free-firefox-account&repo=mozilla_org) have the same, with the exceptions of locales managed outside of Pontoon (ar, de, fr so far).

We should leave it as is.

File: l10n/en/firefox/features/sync-2023.ftl
Comment (flodolo) [reply l10n/en/firefox/features/sync-2023.ftl:18]: Same here: [current string](https://github.com/mozilla-l10n/www-l10n/blob/master/en/firefox/features/sync-2023.ftl#L20), [current translations](https://transvision.flod.org/string/?entity=mozilla_org/en/firefox/features/sync-2023.ftl:features-sync-sign-up-for-a-free-firefox-account&repo=mozilla_org) all using term+account (minus Smartling locales).

File: bedrock/firefox/templates/firefox/browsers/compare/safari.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/browsers/compare/safari.html:246]: Also, should `data-cta-name` now be `"Mozilla account"`?

**Review by alexgibson — APPROVED** [review]
r+wc

File: bedrock/firefox/templates/firefox/browsers/compare/safari.html
Comment (reemhamz) [reply bedrock/firefox/templates/firefox/browsers/compare/safari.html:246]: Ah, good looking out :)

File: bedrock/security/templates/security/third-party-injection-policy.html
Comment (reemhamz) [inline bedrock/security/templates/security/third-party-injection-policy.html:148]: Updated this security page in this recent commit, too.

**Comment by reemhamz** [comment]
@flodolo Done w/ your requested changes. Have a final look and let me know if it's good to go?

**Review by flodolo — APPROVED** [review]
Thanks, LGTM

**Pull request merged**
